pkg/integrations/mysqld_exporter/mysqld-exporter.go

```diff
@@ -0,0 +1,100 @@
+// Package mysqld_exporter embeds https://github.com/prometheus/mysqld_exporter
+package mysqld_exporter //nolint:golint
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/pkg/integrations/config"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/mysqld_exporter/collector"
+)
+
+// Integration is the mysqld_exporter integration. The integration scrapes metrics
+// from a mysqld process.
+type Integration struct {
+	c        Config
+	logger   log.Logger
+	exporter *collector.Exporter
+}
+
+// New creates a new mysqld_exporter integration.
+func New(log log.Logger, c Config) (*Integration, error) {
+	dsn := c.DataSourceName
+	if len(dsn) == 0 {
+		dsn = os.Getenv("MYSQLD_EXPORTER_DATA_SOURCE_NAME")
+	}
+	if dsn == "" {
```

Comment:
question: what's the reasoning behing using `len(dsn) == 0` above and `dsn == ""` here?

Reply:
Sloppiness :) I'll change it so it uses the same check